### PR TITLE
Enable pubsub dynamic destinations by default

### PR DESCRIPTION
**Pull request by reuvenlax**

### CODE REVIEW

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java [hunk 1]
```diff
@@ -155,7 +155,7 @@
 
   // Disable this support for now until the Dataflow backend fully supports this option.
   @Description("Whether to allow dynamic pubsub destinations. Temporary option: will be removed.")
```
Comment (aaltay) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java:157]: Could we just remove the option instead?

**Review by aaltay — APPROVED** [review]

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 2]
```diff
@@ -1918,12 +1918,7 @@ private static void translate(
       } else {
         DataflowPipelineOptions options =
```
Comment (aaltay) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java:1919]: options is unused.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java (hunk 1 above)
Comment (reuvenlax) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java]: done

**Comment by reuvenlax** [comment]
Run Spotless PreCommit

**Comment by reuvenlax** [comment]
Run Spotless PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**